doc/developer_guide.md

```diff
@@ -53,277 +55,366 @@ Create a new class called **MyMapEventListener** which implements MapEventListne
-  {% highlight swift %}COMING SOON...
+  {% highlight swift %}
+  
+public class MyMapEventListener : NTMapEventListener {
+    
+    var oldClickLabel: NTBalloonPopup?
+    
+    var mapView: NTMapView?
+    var vectorDataSource: NTLocalVectorDataSource?
+    
+    convenience init(mapView: NTMapView?, vectorDataSource: NTLocalVectorDataSource?) {
+        
+        self.init()
+        
+        self.mapView = mapView
+        self.vectorDataSource = vectorDataSource
+    }
+    
+    public override func onMapMoved() {
+        
+        // This is called every time the map moves, and we can calculate positions as such:
+        
+        var topLeft = mapView?.screen(toMap: NTScreenPos(x: 0, y: 0))
+        
+        let width: Float = Float((mapView?.frame.width)!)
+        let height: Float = Float((mapView?.frame.width)!)
+        
+        var bottomRight = mapView?.screen(toMap: NTScreenPos(x: width, y: height))
+    }
+    
+    public override func onMapClicked(_ mapClickInfo: NTMapClickInfo!) {
+        
+        // Remove old click label
+        if (oldClickLabel != nil) {
+            vectorDataSource?.remove(oldClickLabel)
+            oldClickLabel = nil
+        }
+        
+        let clickPos = mapClickInfo?.getClickPos()
+        
+        let styleBuilder = NTBalloonPopupStyleBuilder();
+        // Make sure this label is shown on top all other labels
+        styleBuilder?.setPlacementPriority(10)
+        let style = styleBuilder?.buildStyle()
+        
+        // Check the type of the click
+        var clickMsg = ""
+        let clickType = mapClickInfo?.getClickType()
+        
+        if (clickType == NTClickType.CLICK_TYPE_SINGLE) {
+            clickMsg = "Single map click!"
+        } else if (clickType == NTClickType.CLICK_TYPE_LONG) {
+            clickMsg = "Long map click!"
+        } else if (clickType == NTClickType.CLICK_TYPE_DOUBLE) {
+            clickMsg = "Double map click!"
+        } else if (clickType == NTClickType.CLICK_TYPE_DUAL) {
+            clickMsg = "Dual map click!"
+        }
+        
+        let wgs84Clickpos = mapView?.getOptions().getBaseProjection()?.toWgs84(clickPos)
+        let msg = String(describing: wgs84Clickpos?.getY()) + String(describing: wgs84Clickpos?.getX())
+        
+        let clickPopup = NTBalloonPopup(pos: clickPos, style: style, title: clickMsg, desc: msg)
+        
+        vectorDataSource?.add(clickPopup)
+        oldClickLabel = clickPopup
+    }
+}
 
   {% endhighlight %}
   </div>
+
+  <div class="Carousel-item js-Tabpanes-item--lang js-Tabpanes-item--lang--kotlin">
+  {% highlight kotlin %}
   
+class MyMapEventListener(var mapView: MapView?, var vectorDataSource: LocalVectorDataSource?) : MapEventListener() {
+
+    var oldClickLabel: BalloonPopup? = null
+
+    override fun onMapMoved() {
+
+        // This is called every time the map moves, and we can calculate positions as such:
+
```

Comment:
"as such" sounds weird to me. Did you mean such as? Or maybe just omit it all together and say "CARTO calculates following positions:"